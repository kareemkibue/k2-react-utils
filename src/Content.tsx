import * as React from 'react';

/** Interface for Content Props */
interface IContentProps {
	text: string;
}

/** Content Component with HTML sanitization */
class Content extends React.Component<IContentProps, {}> {
	private parseMarkup(rawHtml: string) {
		return { __html: rawHtml };
	}

	render() {
		const { text } = this.props;

		return <span dangerouslySetInnerHTML={this.parseMarkup(text)} />;
	}
}

export { Content, IContentProps };
