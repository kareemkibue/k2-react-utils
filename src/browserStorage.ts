
type BrowserStorageType = 'SESSION' | 'LOCAL'
// type ObjectValue = null | string | Array<any> | { [field: string]: any } 

const getStoredValue = (storageType: BrowserStorageType, key: string): any => {
    if (storageType === "LOCAL") {
        const storedValue = localStorage.getItem(key) || '';
        return getObjectValue(storedValue);
    }
    if (storageType === "SESSION") {
        const storedValue = sessionStorage.getItem(key);
        return getObjectValue(storedValue);
    }

    return;
}

const setValue = (storageType: BrowserStorageType, key: string, value: any) => {
    if (storageType === "LOCAL") {
        localStorage.setItem(key, JSON.stringify(value))
    } if (storageType === "SESSION") {
        sessionStorage.setItem(key, JSON.stringify(value))
    }
}

const clearValue = (storageType: BrowserStorageType, key: string): void => {
    if (storageType === "LOCAL") {
        localStorage.removeItem(key)
    }
    if (storageType === "SESSION") {
        sessionStorage.removeItem(key)
    }
}

const getObjectValue = (value: string | null): any => {
    if (value === null) {
        return null
    }

    try {
        return JSON.parse(value)
    } catch (error) {
        return String(value)
    }
}

const browserStorage = { getStoredValue, setValue, clearValue }

export { browserStorage, BrowserStorageType }