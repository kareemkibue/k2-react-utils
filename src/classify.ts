/**
 * Classify - Returns a list of classes after an Evalution
 */
export function classify(classObject: { [field: string]: boolean }): string {
	let classList: string = '';

	for (const i in classObject) {
		if (classObject[i] === true) {
			classList += i + ' ';
		}
	}

	return classList.trim();
}
