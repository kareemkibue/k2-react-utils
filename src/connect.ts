// import { connect as connectReactRedux } from 'react-redux';

/**
 * react-redux connect decorator implementation for Typescript
 * Suppresses @types/react-redux issue when using connect as a decorator
 */
const connect = (mapStateToProps: any, mapDispatchToProps?: any, mergeProps?: any, options?: any) => {
    try {
        const reactRedux = require('react-redux');

        return (target: any) => (reactRedux.connect(mapStateToProps, mapDispatchToProps, mergeProps, options)(target) as any);
    } catch (error) {
        // console.error(error)
        return
    }
}


export { connect }