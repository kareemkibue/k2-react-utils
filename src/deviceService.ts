
type BrowserName = null | "chrome" | "chromium" | "safari" | "firefox" | "ie" | "ie mobile" | "edge" | "opera" | "opera mini"
type OperatingSytem = null | "windows" | "mac os" | "android" | "ios";
type Platform = "desktop" | "mobile"






const getOperatingSystem = (userAgent: string = navigator.userAgent): OperatingSytem => {
    if (userAgent.match(/Windows|Win64|Win32/i)) {
        return 'windows'
    }
    if (userAgent.match(/iPhone|iPad|iPod/i)) {
        return 'ios'
    }
    if (userAgent.match(/Mac OS X|Mac_PowerPC/i)) {
        return 'mac os'
    }
    if (userAgent.match(/Android/i)) {
        return 'android'
    }

    return null
}

const getBrowserName = (userAgent: string = navigator.userAgent): BrowserName => {
    if (userAgent.match(/edge/i)) {
        return 'edge'
    }
    if (userAgent.match(/chromium/i)) {
        return 'chromium'
    }
    if (userAgent.match(/chrome/i)) {
        return 'chrome'
    }
    if (userAgent.match(/firefox/i)) {
        return 'firefox'
    }
    if (userAgent.match(/msie|trident/i)) {
        return 'ie'
    }
    if (userAgent.match(/IEMobile/i)) {
        return 'ie mobile'
    }
    if (userAgent.match(/safari/i)) {
        return 'safari'
    }
    if (userAgent.match(/Opera Mini/i)) {
        return 'opera mini'
    }
    if (userAgent.match(/opera|opr/i)) {
        return 'opera'
    }

    return null;
};


const getPlatform = (userAgent: string = navigator.userAgent): Platform => {
    if (userAgent.match(/iPhone|iPad|iPod|Android|BlackBerry|IEMobile|Opera Mini/i)) {
        return 'mobile'
    }

    return "desktop"
}






export {
    getBrowserName,

    getOperatingSystem,
    getPlatform, BrowserName, OperatingSytem, Platform
}