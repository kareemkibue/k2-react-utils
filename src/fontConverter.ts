const defaultBrowserFontSize: string = '16px';

const getPixelUnit = (pixelValue: string): number => {
	const pxIndex: number = pixelValue.indexOf('px');
	const pixelUnit: number = Number(pixelValue.slice(0, pxIndex));
	return pixelUnit;
};

/** Font utility that converts pixels to rem units */
const convertPixelsToRem = (pixelValue: string, baseFontSize: string = defaultBrowserFontSize): string => {
	const pixelUnit: number = getPixelUnit(pixelValue);
	const baseFontUnit: number = getPixelUnit(baseFontSize);
	const calculatedRemUnit: number = pixelUnit / baseFontUnit;

	/* console.log('typeof calculatedRemUnit=', typeof calculatedRemUnit);
	console.log('calculatedRemUnit=', calculatedRemUnit);
	console.log('\n==='); */

	// if (typeof calculatedRemUnit !== 'number') {
	/* if (calculatedRemUnit != 'undefined') {
		return String(calculatedRemUnit) + 'rem';
	} else {
		// throw new Error('Invalid pixel unit.');
		return false;
	} */
	// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

	return String(calculatedRemUnit) + 'rem';
};

/** Font utility that converts pixels to em units */
const convertPixelsToEm = (pixelValue: string, baseFontSize: string = defaultBrowserFontSize): string => {
	const pixelUnit: number = getPixelUnit(pixelValue);
	const baseFontUnit: number = getPixelUnit(baseFontSize);
	const calculatedEmUnit: number = pixelUnit / baseFontUnit;

	return String(calculatedEmUnit) + 'em';
};

export { convertPixelsToRem, convertPixelsToEm };
