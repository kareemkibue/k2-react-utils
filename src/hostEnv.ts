const getEnvFromHost = (): { isLocal: boolean; host: string } => {
	const host: string = window.location.host;
	const isLocal: boolean = host.indexOf('localhost') >= 0;

	return {
		isLocal,
		host,
	};
};

export { getEnvFromHost };
