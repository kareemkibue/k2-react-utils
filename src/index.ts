export { convertPixelsToRem, convertPixelsToEm } from './fontConverter';
export { classify } from './classify';
export { connect } from './connect';
export { Content, IContentProps } from './Content';
export { Parser } from './parser';
export { getEnvFromHost } from './hostEnv';
