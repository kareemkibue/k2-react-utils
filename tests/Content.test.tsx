import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Content } from '../src/Content';

test('renders', () => {
	// Act
	const component = renderer.create(<Content text="http://www.facebook.com" />);

	// Assert
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
