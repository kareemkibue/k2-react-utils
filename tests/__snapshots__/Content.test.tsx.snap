// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "http://www.facebook.com",
    }
  }
/>
`;
